### PR TITLE
Feature/streams

### PR DESCRIPTION
**Pull request by aemadrid**
Adds support for all stream related commands in value mode. Mostly porting the code from the ruby driver.

_Note:_ Still need to investigate how to support futures for pipelining.

### CODE REVIEW

**Comment by larskluge** [comment]
I tried the very basics and it's looking good so far. All as expected/known from its Ruby counterpart. 

And built a tiny first project based on it: https://github.com/larskluge/rsone

Will report if I find anything down the road.

**Comment by aemadrid** [comment]
@larskluge thx! I've looked at implementing 1hashify1/etc for pipelines/futures but I cannot seem to get it to work. Not sure how you feel about having streams not working there.

**Comment by stefanwille** [comment]
@aemadrid thanks for the PR. This looks like good start.

If you can get away with using the builtin commands primitives such as `string_command()`, there is a good chance that your implementation works with pipelines/futures too.

I see the spec fails in CI. Is it green for you in local development?

**Comment by aemadrid** [comment]
@stefanwille it does pass locally. I think the problem is that travis doesn't have the right redis version (>= 5). BTW I figured out a solution to the pipeline issue but it will be a much bigger PR. My idea is to `command(args, unwrapper)` where unwrapper is a strategy like :int_or_nil, :string_array, etc. so that could be applied at the right time. Very similar to what the ruby driver does. That should work for pipelines, transactions, etc. and also work for all hash, stream, etc. responses.

**Comment by Xosmond** [comment]
@aemadrid @stefanwille  Is there any news on this PR?

**Comment by stefanwille** [comment]
No

File: src/redis/commands.cr [hunk 1]
```diff
@@ -1713,5 +1862,89 @@ class Redis
...
+      hsh["consumers"] = consumers
+      hsh
+    end
+
+    private def hashify_stream_pending_details(result)
+      Array(Hash(String, String)).new.tap do |ary|
+        if result.is_a? Array(Redis::RedisValue)
+          result.each do |arr|
+            if arr.is_a? Array(RedisValue)
+              ary << {
```
Comment (dscottboggs) [inline src/redis/commands.cr]: This should certainly be a struct

```crystal
struct StreamPendingDetails
  property entry_id : String
  property consumer : String
  property elapsed : String
  property count : String
  def initialize(result : Array(RedisValue)
    @entry_id = result[0].to_s
    @consumer = result[1].to_s
    @elapsed = result[2].to_s
    @count = result[3].to_s
  end
  def self.new_collection(result : Array(RedisValue)) : Array(StreamPendingDetails)
    Array(StreamPendingDetails).new.tap do |ary|
      result.each do |result_ary|
        ary << StreamPendingDetails.new result_ary
      end
    end
  end
end
```
There's no sense in using a hash where all potential keys are known.

**Comment by anykeyh** [comment]
Hello,

I was planning to work precisely on this feature for a project of mine. Very happy that someone already started it <3. 

However, what is currently blocking this from being merged to master? Except for the travis Redis version, is there anything needed to be implemented?

I would be happy to help.

**Comment by kostya** [comment]
I think it mergable, but need update to master (btw method hashify was already added). Then if specs passing, also check that this feature works in real.

**Pull request closed without merging**